coderd/coderd_test.go

```diff
@@ -94,6 +94,8 @@ func TestAuthorizeAllEndpoints(t *testing.T) {
 		t.Cleanup(func() { close(tickerCh) })
 
 		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer t.Cleanup(cancelFunc) // Defer to ensure cancelFunc is executed first.
```

Comment:
Does `defer cancelFunc()` not work here? I would expect defer to run before cleanups.

Reply:
Since this is a helper function, `defer cancelFunc()` would run before the calling test gets a chance to do anything with it. And a `t.Cleanup` without `defer` would cancel the context too late, it would not trigger teardown. So I realize this is a bit of a sneaky way to do it, but this way `t.Cleanup(cancelFunc)` will be left on top of the cleanup stack when this function exits, triggering the teardown we want (and same order as before the refactor).

Reply:
See comment on the other instance: https://github.com/coder/coder/pull/3113/files#r928922910.